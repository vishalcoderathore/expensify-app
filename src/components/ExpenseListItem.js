import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({description, amount, createdAt, note, dispatch, id}) => (
    <div>
        
        <Link to={`/edit/${id}`}><h1>Description: {description}</h1></Link>
        <p>Amount: {amount} ,Created on: {createdAt}</p>
        <p>Expense Note: {note}</p>
       
    </div>
        
);
export default ExpenseListItem;

