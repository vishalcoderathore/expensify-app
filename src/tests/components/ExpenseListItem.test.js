import React from 'react';
import  ExpenseListItem  from '../../components/ExpenseListItem';
import {expenses} from '../fixtures/dummyExpenses';
import { shallow } from 'enzyme';

test('should render expense list item correctly', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[1]}/>);

    expect(wrapper).toMatchSnapshot();
});