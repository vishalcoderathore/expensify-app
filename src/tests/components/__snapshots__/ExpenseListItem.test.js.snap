// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item correctly 1`] = `
<Link
  className="list-item"
  replace={false}
  to="/edit/2"
>
  <div>
    <h3
      className="list-item__title"
    >
      Groceries
    </h3>
    <span
      className="list-item__sub-title"
    >
      December 27th, 1969
    </span>
  </div>
  <div>
    <h3
      className="list-item__title"
    >
      $100.00
    </h3>
  </div>
</Link>
`;
